contrib/experimental/great_expectations_experimental/metrics/column_values_string_integers_monotonically_increasing.py

```diff
@@ -0,0 +1,153 @@
+
+from great_expectations.core import ExpectationConfiguration
+from great_expectations.execution_engine import (
+    ExecutionEngine,
+    PandasExecutionEngine,
+    SparkDFExecutionEngine,
+)
+from great_expectations.execution_engine.execution_engine import (
+    MetricDomainTypes,
+    MetricPartialFunctionTypes,
+)
+from great_expectations.execution_engine.sqlalchemy_execution_engine import (
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.metrics.import_manager import (  # sa,
+    F,
+    Window,
+    sparktypes,
+)
+from great_expectations.expectations.metrics.map_metric_provider import (
+    ColumnMapMetricProvider,
+    column_function_partial,
+    metric_partial,
+)
+from great_expectations.validator.metric_configuration import MetricConfiguration
+
+
+class ColumnValuesStringIntegersMonotonicallyIncreasing(ColumnMapMetricProvider):
+    function_metric_name = "column_values.string_integers.monotonically_increasing"
+
+    @column_function_partial(engine=PandasExecutionEngine)
+    def _pandas(self, data, _metrics, **kwargs):
+
+        try:
+            temp_column = [int(x) for x in data if x.isdigit()]
+        except AttributeError:
+            raise TypeError(
+                "Column must be a string-type capable of being cast to int."
+            )
+
+        if len(temp_column) != len(data):
+            raise TypeError(
+                "Column must be a string-type capable of being cast to int."
+            )
+
+        series_diff = np.diff(temp_column)
+
+        return series_diff >= 0
+
+    @metric_partial(
+        engine=SparkDFExecutionEngine,
+        partial_fn_type=MetricPartialFunctionTypes.WINDOW_FN,
+        domain_type=MetricDomainTypes.COLUMN,
+    )
+    def _spark(
+        cls,
+        execution_engine,
+        metric_domain_kwargs,
+        metric_value_kwargs,
+        metrics,
+        runtime_configuration,
+    ):
+        column_name = metric_domain_kwargs["column"]
+        table_columns = metrics["table.column_types"]
+        column_metadata = [col for col in table_columns if col["name"] == column_name][
+            0
+        ]
+
+        if isinstance(column_metadata["type"], (sparktypes.StringType)):
+            try:
+                column = F.col(column_name).cast(sparktypes.IntegerType())
+            except TypeError:
+                raise TypeError(
+                    "Column must be a string-type capable of being cast to int."
+                )
+        else:
+            raise TypeError(
+                "Column must be a string-type capable of being cast to int."
+            )
+
+        compute_domain_kwargs = metric_domain_kwargs
+
+        (
+            df,
+            compute_domain_kwargs,
+            accessor_domain_kwargs,
+        ) = execution_engine.get_compute_domain(
+            compute_domain_kwargs, domain_type=MetricDomainTypes.COLUMN
+        )
+
+        if any(np.array(df.select(column.isNull()).collect())):
```

Comment:
What is the case here where and error isn't raised in lines 72-82 but is raised here?

Reply:
Casting non-integers to int doesn't error, but resolves to null at collection; this handles that case.